styx-service-common/src/main/java/com/spotify/styx/state/RunState.java

```diff
@@ -249,8 +212,11 @@ public RunState submitted(WorkflowInstance workflowInstance, String executionId,
     public RunState started(WorkflowInstance workflowInstance) {
       switch (state()) {
         case SUBMITTED:
-        case PREPARE:
           return state(RUNNING);
+        case PREPARE:
+          return state(RUNNING, data().builder()
+              .tries(data().tries() + 1)
```

Comment:
I don't know what the reason was. Missing `tries` increase on that transition branch was introduced by https://github.com/spotify/styx/pull/33.

I doubt bypassing `submitting` and `submitted` states would actually work when executing a workflow instance because that would mean missing many things. During replay, this `tries` is not important I think.

My suggestion is we can remove support of this transition.

Reply:
BTW there are a few things in this file marked as `backward compatibility`, e.g. https://github.com/spotify/styx/pull/826/files#diff-05de7b680cea03b57e3a1df7bbbc1258R179 . I think we can also kill those.

Reply:
Cool, thanks for the input. I think we'll do that as part of a following PR just to keep this PR limited in scope.